### PR TITLE
Nav and Auth

### PR DESCRIPTION
**Pull request by asimonson1125**
fork hosted [here](https://devcade-website-asimonson1125fork-devcade-website.apps.okd4.csh.rit.edu/)
Created the core pages and got CSH OIDC working for auth.

### CODE REVIEW

**Comment by asimonson1125** [comment]
OIDC does have a secret key that you'd need to login if you test locally.  Let me know if you need it

File: src/templates/header.html [hunk 1]
```diff
@@ -2,18 +2,65 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link
+      rel="stylesheet"
+      href="{{ url_for('static', filename='css/devcade.css') }}"
+    />
+    <link
+      rel="shortcut icon"
+      href="{{ url_for('static', filename='favicon.ico') }}"
+    />
     <title>Devcade</title>
```
Comment (MTFT-Games) [inline src/templates/header.html]: Would it be possible to make this variable by page instead of baked into the header?

Comment (asimonson1125) [reply src/templates/header.html]: Yeah we could add a block that can be modified by either variables or by templates in the same way the 'background' photo is changed.  I would think that isn't standard though, websites tend to only have a single favicon.

Comment (asimonson1125) [reply src/templates/header.html]: also an easy js method: `document.querySelector("link[rel*='icon']").href = "https://github.githubassets.com/favicons/favicon-dark.png"`

Comment (MTFT-Games) [reply src/templates/header.html]: I was referring to the page title not the favicon. 
I agree the favicon doesn't need to change between pages but the title that shows up as the tab name in the browser I believe commonly does change per page.

Either way I presume the same fix applies

**Pull request merged**